smartsim/_core/_cli/scripts/dragon_install.py

```diff
@@ -77,67 +145,117 @@ def _version_filter(asset_name: str) -> bool:
 
-    # take the most recent release for the given pin
-    assets = releases[0].assets
+    if not releases:
+        release_titles = ", ".join(release.title for release in all_releases)
+        raise SmartSimCLIActionCancelled(
+            f"Unable to find a release for dragon version {request.pkg_version}. "
+            f"Available releases: {release_titles}"
+        )
+
+    # install the latest release of the target version (including pre-release)
+    release = releases[0]
+    assets = list(release.get_assets())
+
+    if not assets:
+        raise SmartSimCLIActionCancelled(
+            f"Unable to find assets for dragon release {release.title}"
+        )
 
     return assets
 
 
-def filter_assets(assets: t.Collection[GitReleaseAsset]) -> t.Optional[GitReleaseAsset]:
+def filter_assets(
+    request: DragonInstallRequest, assets: t.Collection[GitReleaseAsset]
+) -> t.Optional[GitReleaseAsset]:
     """Filter the available release assets so that HSTA agents are used
     when run on a Cray EX platform
 
+    :param request: details of a requesst for the installation of the dragon package
     :param assets: The collection of dragon release assets to filter
     :returns: An asset meeting platform & version filtering requirements"""
     # Expect cray & non-cray assets that require a filter, e.g.
     # 'dragon-0.8-py3.9.4.1-bafaa887f.tar.gz',
     # 'dragon-0.8-py3.9.4.1-CRAYEX-ac132fe95.tar.gz'
-    asset = next(
-        (
-            asset
-            for asset in assets
-            if _version_filter(asset.name)
-            and _platform_filter(asset.name)
-            and _pin_filter(asset.name)
-        ),
-        None,
+    all_assets = [asset.name for asset in assets]
+
+    assets = list(
+        asset
+        for asset in assets
+        if _version_filter(asset.name) and _pin_filter(asset.name, request.pkg_version)
     )
+
+    if len(assets) == 0:
+        available = "\n\t".join(all_assets)
+        logger.warning(
+            f"Please specify a dragon version (e.g. {DEFAULT_DRAGON_VERSION}) "
+            f"of an asset available in the repository:\n\t{available}"
+        )
+        return None
+
+    asset: t.Optional[GitReleaseAsset] = None
+
+    # Apply platform filter if we have multiple matches for python/dragon version
+    if len(assets) > 0:
+        asset = next((asset for asset in assets if _platform_filter(asset.name)), None)
+
+    if not asset:
+        asset = assets[0]
+        logger.warning(f"Platform-specific package not found. Using {asset.name}")
+
     return asset
 
 
-def retrieve_asset_info() -> GitReleaseAsset:
+def retrieve_asset_info(request: DragonInstallRequest) -> GitReleaseAsset:
     """Find a release asset that meets all necessary filtering criteria
 
-    :param dragon_pin: identify the dragon version to install (e.g. dragon-0.8)
+    :param request: details of a requesst for the installation of the dragon package
```

Comment:
`requesst` -> `request`

Reply:
it's in a couple places

Reply:
I'm channeling @MattToast here!

Reply:
I the reader gets the gist, its good enough in my book! 😉